plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -224,7 +224,8 @@ public void dropNamespace(ConnectorSession session, String namespace)
     {
         // basic sanity check to provide a better error message
         if (!listTables(session, Optional.of(namespace)).isEmpty() ||
-                !listViews(session, Optional.of(namespace)).isEmpty()) {
+                !listViews(session, Optional.of(namespace)).isEmpty() ||
+                !listMaterializedViews(session, Optional.of(namespace)).isEmpty()) {
```

Comment:
Doesn't belong to "Move the business logic for listing tables into the IcebergMetadata"